src/connection.js

```diff
@@ -92,6 +94,11 @@ module.exports = function connection (swarm) {
       })
     },
 
+    relay (config) {
+      swarm.relay = true
+      swarm.transport.add(Circuit.tag, new Circuit.Dialer(swarm, config))
```

Comment:
I rather prefer to make relay specific and not a 'transport' like this. The transport interface will evolve and that doesn't necessarily need to trickle to relay. 

On the dial.js, after trying all transports, then it should try relay to get that first conn.